reference/src/glossary.md

```diff
@@ -73,10 +73,44 @@ niches. For example, the "all bits uninitialized" is an invalid bit-pattern for
 `&mut T`, but this bit-pattern cannot be used by layout optimizations, and is not a
 niche.
 
+#### Place
+
+A *place* (called "lvalue" in C and "glvalue" in C++) is the result of computing a [*place expression*][place-value-expr].
+A place definitely stores some location in memory (a "pointer"), and might contain more information such as size or alignment (the details will have to be determined as the Rust Abstract Machine gets specified more precisely).
+A place has a type.
+Places are ephemeral; they arise during the computation of an instruction but cannot be persisted in memory.
+
+The key operations on a place are:
+* storing a [value](#value) of the same type in it (when it is used on the left-hand side of an assignment),
+* turning it into a [pointer value](#value) (when it is used inside `&expr`),
+* and loading a [value](#value) of the same type from it (through the place-to-value coercion).
+
+
+#### Value
+
+A *value* (called "value of the expression" or "rvalue" in C and "prvalue" in C++) is what gets stored in a [place](#place), and also the result of computing a [*value expression*][place-value-expr].
+A value has a type.
+For example, a value of type `u8` is a mathematical integer in the range `0..256`.
+Values can be (according to their type) turned into a list of bytes, which is called a [representation](#representation) of the value.
+Values are ephemeral; they arise during the computation of an instruction but are only ever persisted in memory through their representation.
```

Comment:
Does "ephemeral" fit here? Is there anything to a value other than its byte representation which exists during a computation and then doesn't get persisted in memory?

Reply:
Pointer provenance maybe ? AFAICT that gets persisted in memory but it is not part of their representation, although we haven't clarified this.

Reply:
The provenance of a pointer is part of `Pointer`.

Reply:
I think this is made clearer below, where it is shown that bytes can belong to a pointer, and therefore provenance is part of the representation.

Reply:
The difference between a value and its representation is comparable to the difference between run-time data and serialized data. The run-time data is also ephemeral in some sense; only serialized data is persisted across machine reboots, for example.